src/pages/index.js

```diff
@@ -17,47 +42,111 @@ popupImage.setEventListeners();
 const openPopupImage = (data) => {
   popupImage.open(data);
 };
-const renderCard = (card) => {
-  cardRenderer.addItem(createCard(card));
+
+const deletePopup = new PopupWithForm(constants.deletePopup, (data) => {
+  api
+    .deleteCard(data._id, data)
+    .then(() => {
+      deletePopup.submitButton.textContent = "Delete Card";
+      cardArray = document.querySelectorAll(".card");
+      cardArray.forEach((card) => {
```

Comment:
`      cardArray = document.querySelectorAll(".card");
`

You don't need to search all cards. Please, search only once certain card using `id`

It's a waste of resources